arelle/plugin/validate/NL/rules/fr_nl.py

```diff
@@ -0,0 +1,57 @@
+"""
+See COPYRIGHT.md for copyright information.
+"""
+from __future__ import annotations
+
+import re
+from typing import Any, Iterable
+
+from arelle import ModelDocument
+from arelle.ValidateXbrl import ValidateXbrl
+from arelle.typing import TypeGetText
+from arelle.utils.PluginHooks import ValidationHook
+from arelle.utils.validate.Decorator import validation
+from arelle.utils.validate.Validation import Validation
+from ..DisclosureSystems import (
+    DISCLOSURE_SYSTEM_NT16,
+    DISCLOSURE_SYSTEM_NT17,
+    DISCLOSURE_SYSTEM_NT18,
+)
+from ..PluginValidationDataExtension import PluginValidationDataExtension
+
+_: TypeGetText
+
+
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+    disclosureSystems=[
+        DISCLOSURE_SYSTEM_NT16,
+        DISCLOSURE_SYSTEM_NT17,
+        DISCLOSURE_SYSTEM_NT18,
+    ],
+)
+def rule_fr_nl_2_06(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation] | None:
+    """
+    FR-NL-2.06: A CDATA section MAY NOT be used. It is not allowed to include a CDATA section in an XBRL instance document.
+    """
+    pattern = re.compile(r"<!\[CDATA\[")
+    modelXbrl = val.modelXbrl
+    for doc in modelXbrl.urlDocs.values():
+        if doc.type == ModelDocument.Type.INSTANCE:
+            # Default etree parsing strips CDATA from value.
+            # Load document line by line and check raw value against regex.
```

Comment:
Good point on the regex. I've updated the pattern and comment accordingly.

Since whatever is processing the SOAP request wouldn't really care if the contents of the CDATA block are XML or not (correct me if I'm wrong), I figure running the text through the XML parser doesn't add anything to the validation and only introduces potential edge cases that the parser might be quietly erasing.

Reply:
> running the text through the XML parser doesn't add anything to the validation

In theory, someone could claims that Arelle is wrong to report an error for `<!-- <![CDATA[ -->` or `<!-- ]]> -->`, and they would be correct: we're implementing a "willful violation" of the manual because it's complicated to implement as written and we think what we're better implementing their intent (ensure `"<![CDATA["+instance+"]]>"` does not cause the outer XML parsing to fail).

Reply:
Right, the quoted part of my comment was referring to the idea that running `instance` through an `XMLParser` doesn't help us identify if `"<![CDATA["+instance+"]]>"` would cause outer XML parsing to fail, we just need to know if the end sequence appears in the text, regardless of the text's XML validity or parse-ability.